tests/server/test_workon.py

```diff
@@ -1193,3 +1294,160 @@ async def test_workon_sets_pip_config(
             " the shell. This ensures the proper permission checks are performed."
         ),
     )
+
+
+@pytest.mark.parametrize("iteration", range(1000))
```

Comment:
This test case triggers a compile but it never waits until the compile is finished. This means that we might be branches in this test case that we never test. It might for example happen that none of the 1000 iterations contain a finished compile report. Maybe we have to introduce an additional parameter so that we wait for a compile to finish is some of these 1000 iterations.

Reply:
Sorry, it was a leftover 🤦🏼‍♂️